### PR TITLE
Refarctored FetchablePartitionResponse to use proc macro and add toRecords.

### LINKED ISSUE #49: Add ToRecords for FetchablePartitionResponse
Getting the actual messages out of a [fetch in `PartitionConsumer`](https://github.com/infinyon/fluvio-client-node/blob/e6085a6441931cd953c4b3704a0cc3769e363e40/src/index.ts#L839) is a bit of a pain and pretty much requires doing:
```typescript
const batchesToRecords = (response: FetchablePartitionResponse) => {
    let records = new Array();
    response.records.batches.forEach(batch => {
        batch.records.forEach(record => {
            records.push(record.value);
        });
    })
    return records;
};
```

Let's add something like `toRecords` to the `FetchablePartitionResponse` body which returns an `Array<string>` or something.

### PR DESCRIPTION
**Pull request by simlay**
* Also added to_records on the rest side. Closes #49.
* I'd also like to use the changes in infinyon/node-bindgen#122 so that we can return `Options` for node-bindgen returns and use the `?` operator.

### CODE REVIEW

**Review by sehz — APPROVED** [review]

File: src/consumer.rs [hunk 1]
```diff
@@ -173,54 +165,129 @@ impl JSValue<'_> for OffsetWrapper {
...
+        aborted_transactions
+    }
 
-        let record_set = RecordSetWrapper(self.0.records).try_to_js(js_env)?;
+    #[node_bindgen(getter)]
+    fn records(&'static self) -> Option<RecordSetWrapper> {
+        Some(RecordSetWrapper(&self.0.as_ref()?.records))
+    }
+    #[node_bindgen]
+    fn to_records(&'static self) -> Vec<String> {
```
Comment (nicholastmosher) [inline src/consumer.rs:226]: I'm seeing a bunch of this `&'static self` which smells really fishy to me. Why does this have a `'static` lifetime? How is this method used, and how is the underlying object created? I'm having a very difficult time imagining that this lifetime is actually correct. If the object being referenced is not really truly static (as in `STATIC` or `lazy_static!`) then I think this is a recipe for memory errors.

Could you walk me through the intended usage of this API?

**Review by nicholastmosher — CHANGES_REQUESTED** [review]
I want to talk about this `'static` lifetime that's floating around before this gets merged. I know this code is closely tied to `node_bindgen`, and a quick look at that tells me there is `unsafe` everywhere, so I highly suspect we have a leaky abstraction. `unsafe` allows you to conjure any lifetime out of thin air, so it's our responsibility to make sure we only do that correctly, and if this compiles then I strongly suspect we have not done it correctly.

**Comment by nicholastmosher** [comment]
I did a little bit of digging and I found this code from `nj_core` which I think is incorrect:

https://github.com/infinyon/node-bindgen/blob/master/nj-core/src/basic.rs#L354-L372

```rust
impl JsEnv {
    ...
    pub fn unwrap<T>(&self, js_this: napi_value) -> Result<&'static T, NjError> {
        let mut result: *mut ::std::os::raw::c_void = ptr::null_mut();
        napi_call_result!(crate::sys::napi_unwrap(self.0, js_this, &mut result))?;

        Ok(unsafe {
            let rust_ref: &T = &mut *(result as *mut T);
            rust_ref
        })
    }

    pub fn unwrap_mut<T>(&self, js_this: napi_value) -> Result<&'static mut T, NjError> {
        let mut result: *mut ::std::os::raw::c_void = ptr::null_mut();
        napi_call_result!(crate::sys::napi_unwrap(self.0, js_this, &mut result))?;

        Ok(unsafe {
            let rust_ref: &mut T = &mut *(result as *mut T);
            rust_ref
        })
    }
}
```

JsEnv seems to be a wrapper around a raw pointer. The problem with this is that `JsEnv` is not going to be live for the `'static` lifetime, so this signature is lying to the compiler. This is possible because the compiler will let you assign any lifetime in an expression like `unsafe { let rust_ref: &T = &mut *(result as *mut T); }` (i.e. the `&` of `&T` may be any lifetime). 

What we _do_ want is for the returned reference to live no longer than the wrapper around our raw pointer (basically, we want Rust lifetime rules for our raw pointer). I think that a correct definition for these functions would be more like this:

```rust
impl JsEnv {
    ...
    pub fn unwrap<'a, T>(&'a self, js_this: napi_value) -> Result<&'a T, NjError> {
        let mut result: *mut ::std::os::raw::c_void = ptr::null_mut();
        napi_call_result!(crate::sys::napi_unwrap(self.0, js_this, &mut result))?;

        Ok(unsafe {
            let rust_ref: &T = &mut *(result as *mut T);
            rust_ref
        })
    }

    pub fn unwrap_mut<'a, T>(&'a self, js_this: napi_value) -> Result<&'a mut T, NjError> {
        let mut result: *mut ::std::os::raw::c_void = ptr::null_mut();
        napi_call_result!(crate::sys::napi_unwrap(self.0, js_this, &mut result))?;

        Ok(unsafe {
            let rust_ref: &mut T = &mut *(result as *mut T);
            rust_ref
        })
    }
}
```

I realize that the `nj_core` problems here are out of scope for this issue, but for the meantime we should do as much as we can to contain this `'static` leak by intentionally avoiding using `'static` lifetimes when interacting with `napi` items.

File: src/consumer.rs [hunk 2]
```diff
@@ -173,63 +167,120 @@ impl JSValue<'_> for OffsetWrapper {
...
+                if let Ok(value) = String::from_utf8(value) {
+                    records.push(value);
+                }
+            }
+        }
+        records
+    }
+}
+pub struct AbortedTransactionWrapper<'a>(&'a AbortedTransaction);
+impl TryIntoJs for AbortedTransactionWrapper<'_> {
```
Comment (simlay) [inline src/consumer.rs:252]: This is a work around for infinyon/node-bindgen#124. I'm not in love with it but want to ship this pr.

**Review by sehz — APPROVED** [review]

**Pull request merged**